sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -501,6 +503,20 @@ def send(self, request):
                         self.sleep(retry_settings, request.context.transport)
                         continue
                 break
+            except ServiceResponseError as err:
+                if hasattr(err, "inner_exception") and isinstance(
+                        err.inner_exception, requests.exceptions.ConnectionError):
+                    retries_remaining = self.increment(
+                        retry_settings, request=request.http_request, error=err)
+                    if retries_remaining:
+                        retry_hook(
+                            retry_settings,
+                            request=request.http_request,
+                            response=None,
+                            error=err)
+                        self.sleep(retry_settings, request.context.transport)
+                        continue
+                    raise err
             except AzureError as err:
```

Comment:
Note: ServiceResponseError is an AzureError, so I think it is handled already. will sync offline about this